tests/integration/test_npm.py

```diff
@@ -30,6 +30,15 @@
             ),
             id="npm_lockfile3_yarn_registry",
         ),
+        pytest.param(
+            utils.TestParameters(
+                repo="https://github.com/cachito-testing/npm-dev-optional-peer-deps.git",
+                ref="d776f19ab6808dcb1c8e12e3d71322cb664957ce",
+                packages=({"path": ".", "type": "npm"},),
+                check_vendor_checksums=False,
+            ),
+            id="npm_dev_optional_peer_deps",
+        ),
     ],
 )
 def test_npm_smoketest(
```

Comment:
We wanted the test to be e2e I think (external deps are involved)

Reply:
Done.